### PR TITLE
added sales and dashboard screens

### PR DESCRIPTION
**Pull request by DariaMitrofanovna**

### CODE REVIEW

File: ui/src/components/widgets/Funnel/Funnel.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        itemName: { style: { fontSize: 14, fontFamily: 'openSans_regular' } },
+        marker: (name, index) => ({
+            symbol: 'circle',
+            style: {
+                fill: color[index]
+            }
+        })
+    }
+    const label: Label = {
+        content: data1 => data1.value,
```
Comment (achabalov) [inline ui/src/components/widgets/Funnel/Funnel.tsx:37]: I suggest renaming the data

File: ui/src/components/DashboardCard/DashboardCard.module.css [hunk 2]
```diff
@@ -0,0 +1,9 @@
+@value colors "../../assets/styles/theme.module.css";
+
+@value main-light-mode from colors;
```
Comment (achabalov) [inline ui/src/components/DashboardCard/DashboardCard.module.css:3]: the variable is not used

**Review by achabalov — CHANGES_REQUESTED** [review]
view comment

**Review by DRITE — CHANGES_REQUESTED** [review]

File: ui/src/components/widgets/Funnel/Funnel.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+function Funnel({ meta }: FunnelProps) {
+    const { bcName } = meta
+    const { data } = useSelector((state: AppState) => {
+        const cursor = state.screen.bo.bc[bcName]?.cursor
+        return { data: state.data[bcName]?.find(i => i.id === cursor)?.[meta.options.dataKey] as MultivalueSingleValue[] }
+    })
+    const sortedData = data?.sort(sorter)
+    const color = sortedData?.map(i => i.options?.hint) as Array<string>
+    const funnelData = useMemo(() => sortedData?.map(i => ({ ...i, value: Number.parseInt(i.value, 10) })), [data])
```
[the remark this answers is not in the document]
Comment (DariaMitrofanovna) [reply ui/src/components/widgets/Funnel/Funnel.tsx]: fixed

File: ui/src/components/widgets/Funnel/Funnel.tsx (hunk 1 above)
Comment (DariaMitrofanovna) [reply ui/src/components/widgets/Funnel/Funnel.tsx]: fixed

File: ui/src/components/widgets/RingProgress/RingProgress.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        descriptionFieldValue.data = ''
+    }
+    return (
+        <div>
+            <h2 className={styles.header}>{meta?.title}</h2>
+            <div className={styles.container}>
+                <div className={styles.text_container}>
+                    <p className={styles.text1}>{text}</p>
+                    <p className={styles.text2}>{numberFieldValue.data}</p>
+                    <p className={styles.text3}>{descriptionFieldValue.data}</p>
```
[the remark this answers is not in the document]
Comment (DariaMitrofanovna) [reply ui/src/components/widgets/RingProgress/RingProgress.tsx]: fixed

File: ui/src/components/DashboardCard/DashboardCard.module.css (hunk 2 above)
Comment (DariaMitrofanovna) [reply ui/src/components/DashboardCard/DashboardCard.module.css]: fixed

File: ui/src/components/widgets/RingProgress/RingProgress.tsx [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    })
+    const percentFieldValue = useSelector((state: AppState) => {
+        return { data: state.data[bcName]?.[0]?.[meta.options.percentField] }
+    })
+    let percent
+    if (percentFieldValue.data) {
+        percent = parseFloat(percentFieldValue.data as string)
+    } else {
+        percent = 0
+    }
```
[the remark this answers is not in the document]
Comment (DariaMitrofanovna) [reply ui/src/components/widgets/RingProgress/RingProgress.tsx]: fixed

File: ui/src/components/widgets/DashboardList/DashboardList.tsx [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+import styles from '../DashboardList/DashboardList.module.css'
+
+interface DashboardListProps {
+    meta: WidgetTableMeta
+}
+
+function DashboardList({ meta }: DashboardListProps) {
+    return (
+        <div>
+            <h2 className={styles.header}>{meta?.title}</h2>
```
[the remark this answers is not in the document]
Comment (DariaMitrofanovna) [reply ui/src/components/widgets/DashboardList/DashboardList.tsx]: fixed

File: ui/src/components/widgets/RingProgress/RingProgress.module.css [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+  font-size: 14px;
+  line-height: 20px;
+  color: #D9D9D9;
+}
+
+.text2 {
+  font-family: 'openSans_bold';
+  font-size: 20px;
+  line-height: 28px;
+  color: #141F35;
```
[the remark this answers is not in the document]
Comment (DariaMitrofanovna) [reply ui/src/components/widgets/RingProgress/RingProgress.module.css]: fixed

File: ui/src/components/widgets/RingProgress/RingProgress.module.css [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+  font-family: 'openSans_bold';
+  font-size: 20px;
+  line-height: 28px;
+}
+
+.text1 {
+  font-family: 'openSans_regular';
+  font-size: 14px;
+  line-height: 20px;
+  color: #D9D9D9;
```
[the remark this answers is not in the document]
Comment (DariaMitrofanovna) [reply ui/src/components/widgets/RingProgress/RingProgress.module.css]: fixed

File: ui/src/components/widgets/RingProgress/RingProgress.module.css [hunk 9]
```diff
@@ -0,0 +1,40 @@
+.container {
+  display: flex;
+  flex-direction: row;
+  justify-content: flex-start;
+  align-items: center;
+  max-height: 283.25px;
+  max-width: 400px;
```
[the remark this answers is not in the document]
Comment (DariaMitrofanovna) [reply ui/src/components/widgets/RingProgress/RingProgress.module.css]: fixed

File: ui/src/components/widgets/RingProgress/RingProgress.tsx [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+                        width={180}
+                        progressStyle={{
+                            lineCap: 'round',
+                            fillOpacity: 100
+                        }}
+                        innerRadius={0.7}
+                        statistic={{
+                            title: false,
+                            content: { style: { fontFamily: 'openSans_bold', fontSize: '20px', color: '#141F35' } }
+                        }}
```
[the remark this answers is not in the document]
Comment (DariaMitrofanovna) [reply ui/src/components/widgets/RingProgress/RingProgress.tsx]: fixed

File: ui/src/components/widgets/RingProgress/RingProgress.tsx [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+    if (!descriptionFieldValue.data) {
+        numberFieldValue.data = ''
+        descriptionFieldValue.data = ''
+    }
+    return (
+        <div>
+            <h2 className={styles.header}>{meta?.title}</h2>
+            <div className={styles.container}>
+                <div className={styles.text_container}>
+                    <p className={styles.text1}>{text}</p>
```
[the remark this answers is not in the document]
Comment (DariaMitrofanovna) [reply ui/src/components/widgets/RingProgress/RingProgress.tsx]: fixed

File: src/main/resources/meta/widget/dashboard/salesRingProgress.widget.json [hunk 12]
```diff
@@ -0,0 +1,13 @@
+{
+  "name": "salesRingProgress",
+  "title": "Closed Sales",
+  "type": "RingProgress",
+  "bc": "salesRingProgress",
+  "options" : {
```
[the remark this answers is not in the document]
Comment (DariaMitrofanovna) [reply src/main/resources/meta/widget/dashboard/salesRingProgress.widget.json]: fixed

File: src/main/resources/meta/widget/dashboard/salesFunnel.widget.json [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+  "bc": "salesFunnel",
+  "fields": [
+    {
+      "title": "",
+      "key": "salesStatistics",
+      "type": "multivalue"
+    }
+  ],
+  "options": {
+    "dataKey": "salesStatistics",
```
[the remark this answers is not in the document]
Comment (DariaMitrofanovna) [reply src/main/resources/meta/widget/dashboard/salesFunnel.widget.json]: fixed

File: src/main/resources/meta/widget/dashboard/salesRingProgress.widget.json [hunk 14]
```diff
@@ -0,0 +1,13 @@
+{
+  "name": "salesRingProgress",
+  "title": "Closed Sales",
+  "type": "RingProgress",
+  "bc": "salesRingProgress",
```
[the remark this answers is not in the document]
Comment (DariaMitrofanovna) [reply src/main/resources/meta/widget/dashboard/salesRingProgress.widget.json:5]: fixed

File: ui/src/interfaces/widget.ts [hunk 15]
```diff
@@ -17,3 +21,13 @@ export interface StepsWidgetMeta extends WidgetMeta {
...
+
+export interface FunnelWidgetMeta extends WidgetMeta {
+    type: CustomWidgetTypes.Funnel
+    options: WidgetOptions & { dataKey: string; color: { id: string; color: string }[] }
+}
+
+export interface RingProgressWidgetMeta extends WidgetMeta {
+    type: CustomWidgetTypes.RingProgress
+    options: WidgetOptions & { text: string; numberField: string; descriptionField: string; percentField: string }
+}
```
[the remark this answers is not in the document]
Comment (DariaMitrofanovna) [reply ui/src/interfaces/widget.ts]: fixed

File: ui/src/components/DashboardCard/DashboardCard.module.css [hunk 16]
```diff
@@ -0,0 +1,9 @@
...
+
+.container {
+  min-width: 440px;
```
[the remark this answers is not in the document]
Comment (DariaMitrofanovna) [reply ui/src/components/DashboardCard/DashboardCard.module.css]: fixed

**Pull request merged**